### PR TITLE
Fix failing EDA Unit Tests that fail only when installed and run from a K12 org.

### LINKED ISSUE #1088: Failing EDA Unit Tests that fail only when installed and run from a K12 org.
**Note: FOR INTERNAL USE ONLY. For support questions, please use the EDA Chatter group in [The Power of Us Hub](https://powerofus.force.com).
For feature requests, please use the Ideas tab in [The Power of Us Hub](https://powerofus.force.com).**

* **Do you want to request a *feature* or report a *bug*?**
Bug

* **What is the current behavior?**
there are 4 failing EDA Unit Tests that fail only when installed and run from a K12 org.
The unit tests are all in the AFFL_AccRecordType_TEST.cls (addRecordTypeToAcc, checkNPE, fillInPrimaryFieldForContacts and updateAdminAccRecordTypeBulk).

* **What is the expected behavior?**
All the unit tests should pass when run from both EDA and K-12

* **What is the motivation / use case for changing the behavior?**
Ensure functionality works

* **Acceptance Criteria**

  - **Given**: I have k12 installed
  - **When**: I run all unit tests
  - **Then**: All unit tests should pass

### PR DESCRIPTION
**Pull request by skosgi1**
# Critical Changes

# Changes
We've updated the way Affiliation Mappings works when EDA orgs are created for new customers. Now, Affiliations Mappings will use API names for Account record types instead of using labels. We've also updated several Affiliations TDTM classes to accommodate this difference. With these changes, EDA unit tests will pass when run on orgs that also have the K-12 Architecture Kit installed. 

# Issues Closed
#1088 

# New Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: src/classes/AFFL_ContactChange_TDTM.cls [hunk 1]
```diff
@@ -102,23 +105,23 @@ public with sharing class AFFL_ContactChange_TDTM extends TDTM_Runnable {
 
 	                   	//CONTACT FIELD CHANGED
 	                   	if(affl.Primary__c && affl.Contact__c != null && afflOld.Contact__c != null
-	                   		&& affl.Contact__c != afflOld.Contact__c && newRelatedContact != null && oldRelatedContact != null && lookupFieldName != null) {
+	                   		&& affl.Contact__c != afflOld.Contact__c && newRelatedContact != null && oldRelatedContact != null && lookupFieldAPIName != null) {
```
Comment (lifewithryan) [inline src/classes/AFFL_ContactChange_TDTM.cls]: some questions here: same as before with the checking for null and then checking if they're aren't equal, if the old contact field is null, and we are changing to not be null, do we need to allow for that?  Secondly, if affOld.Contact is essentially oldReleatedContact, do we need both of those comparisons? I see that one is the contact itself and one is the contact Id, so I think that case could we just use newRelated and oldRelated and drop the other two?

File: src/classes/AFFL_MultiRecordTypeMapper.cls [hunk 2]
```diff
@@ -159,16 +176,22 @@ public with sharing class AFFL_MultiRecordTypeMapper {
     * @return String The name of the key affiliation Contact field.
     ********************************************************************************************************/
 	public String getKeyAfflFieldByAccRecordType(ID recordTypeId) {
-		for(String recTypeName : accRecTypeToContactLabel.keySet()) {
-            if (Schema.Sobjecttype.Account.getRecordTypeInfosByName().get(recTypeName) != null)
-            {
+		for (String recTypeName : accRecTypeToContactLabel.keySet()) {
```
Comment (lifewithryan) [inline src/classes/AFFL_MultiRecordTypeMapper.cls]: the spacing is off here, in vs code looks like these are tabs and not spaces?

File: src/classes/STG_InstallScript.cls [hunk 3]
```diff
@@ -142,12 +142,12 @@ global without sharing class STG_InstallScript implements InstallHandler {
...
-            mappings.add(new Affl_Mappings__c(Name = 'Household Account', Account_Record_Type__c = 'Household Account', Primary_Affl_Field__c = 'Primary Household'));
-            mappings.add(new Affl_Mappings__c(Name = 'Educational Institution', Account_Record_Type__c = 'Educational Institution', Primary_Affl_Field__c = 'Primary Educational Institution'));
-            mappings.add(new Affl_Mappings__c(Name = 'University Department', Account_Record_Type__c = 'University Department', Primary_Affl_Field__c = 'Primary Department'));
-            mappings.add(new Affl_Mappings__c(Name = 'Sports Organization', Account_Record_Type__c = 'Sports Organization', Primary_Affl_Field__c = 'Primary Sports Organization'));
+            mappings.add(new Affl_Mappings__c(Name = 'Academic Program', Account_Record_Type__c = 'Academic_Program', Primary_Affl_Field__c = 'Primary Academic Program', Auto_Program_Enrollment__c = true, Auto_Program_Enrollment_Status__c = 'Current', Auto_Program_Enrollment_Role__c = 'Student'));
```
Comment (lifewithryan) [inline src/classes/STG_InstallScript.cls]: with this fix in place, do we still need/want to do the record types as API Names as well? (I think with these modifications they should work either way correct? Can users also use the field api names in Primary_Affl_Field__c (such as Primary_Academic_Program__c)? If so, do we want to change that now?

Comment (skosgi1) [reply src/classes/STG_InstallScript.cls]: I think we have W-037866 for that.

File: src/classes/AFFL_AccChange_TDTM.cls [hunk 4]
```diff
@@ -95,29 +103,29 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
 
                    		//ACCOUNT FIELD CHANGED
                    		if(affl.Primary__c && affl.Account__c != null && afflOld.Account__c != null
-                       		&& affl.Account__c != afflOld.Account__c && lookupFieldName != null) {
+                       		&& affl.Account__c != afflOld.Account__c && lookupFieldAPIName != null) {
```
[the remark this answers is not in the document]
Comment (skosgi1) [reply src/classes/AFFL_AccChange_TDTM.cls]: I have actually just changed the label name here, I am not entirely sure how the logic works. Please let me know if you want me to dig deeper.

File: src/classes/AFFL_ContactChange_TDTM.cls (hunk 1 above)
Comment (skosgi1) [reply src/classes/AFFL_ContactChange_TDTM.cls]: I have actually just changed the label name here, I am not entirely sure how the logic works. Please let me know if you want me to dig deeper.

File: src/classes/AFFL_MultiRecordTypeMapper.cls (hunk 2 above)
Comment (skosgi1) [reply src/classes/AFFL_MultiRecordTypeMapper.cls]: Fixed it.

File: src/classes/AFFL_AccChange_TDTM.cls [hunk 5]
```diff
@@ -95,29 +103,29 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
...
-                               		relatedContact.put(lookupFieldName, affl.Account__c);
+                           		if (lookupFieldAPINameOld != null) {
+                               		relatedContact.put(lookupFieldAPINameOld, null);
+                               		relatedContact.put(lookupFieldAPIName, affl.Account__c);
                            		}
                        		}
                        		dmlWrapper.objectsToUpdate.add(relatedContact);
 
-                       		//afflMulti.uncheckOtherPrimariesSameType(); --> we don't need to call this because
+                       		  //afflMulti.uncheckOtherPrimariesSameType(); --> we don't need to call this because
```
[the remark this answers is not in the document]
Comment (skosgi1) [reply src/classes/AFFL_AccChange_TDTM.cls]: Fixed it.

File: src/classes/AFFL_AccChange_TDTM.cls [hunk 6]
```diff
@@ -72,15 +72,23 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
...
+               		String lookupLabelOrAPIName = AFFL_MultiRecordType_TDTM.afflMapper.accRecTypeToContactLabel.get(affl.Affiliation_Type__c);
+                  if (lookupLabelOrAPIName == NULL) {
+                      String affiliationTypeAPI = afflMapper.accRecTypeLabelToAPI.get(affl.Affiliation_Type__c);
+                      lookupLabelOrAPIName = afflMapper.accRecTypeToContactLabel.get(affiliationTypeAPI);
+                  }
+           			  String lookupFieldAPIName = AFFL_MultiRecordType_TDTM.afflMapper.contactLabelNames.get(lookupLabelOrAPIName);
 
                		Affiliation__c afflOld;
-               		String lookupFieldLabelOld, lookupFieldNameOld;
+               		String lookupFieldLabelOrAPIOld, lookupFieldAPINameOld;
```
[the remark this answers is not in the document]
Comment (skosgi1) [reply src/classes/AFFL_AccChange_TDTM.cls]: Fixed it.

File: src/classes/STG_InstallScript.cls (hunk 3 above)
Comment (lifewithryan) [reply src/classes/STG_InstallScript.cls]: oh I thought that work item was strictly for the UI to reflect the label when viewing and the API Name when editing, but if that bit is included, no worries.

File: src/classes/AFFL_AccChange_TDTM.cls (hunk 4 above)
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/classes/AFFL_AccChange_TDTM.cls]: I think that is fine, it's not hurting anything other than some readability.

File: src/classes/AFFL_ContactChange_TDTM.cls (hunk 1 above)
Comment (lifewithryan) [reply src/classes/AFFL_ContactChange_TDTM.cls]: well it certainly wouldn't hurt to gain some understanding at a bit lower level, but it's not something I would hold up a merge for.

File: src/classes/STG_InstallScript.cls (hunk 3 above)
Comment (skosgi1) [reply src/classes/STG_InstallScript.cls]: Yep, we decided to make that change in that WI. Resolving.

**Comment by dgamboni** [comment]
Tested / Passes QE. @juliesnow please give love to our RN's! Thanks!

**Pull request merged**